lib/tic_tac_toe.rb

```diff
@@ -1,15 +1,52 @@
 require_relative './command_line_in'
 require_relative './prompt'
 require_relative './board_presenter'
+require_relative './board'
+require_relative './input_validator'
 
 class TicTacToe
-    def play
+    @@selected_token
+    @@selected_move
+    
+    attr_accessor :board, :board_presenter, :input_validator
+
+    def initialize
+        @board = Board.new
+        @board_presenter = BoardPresenter.new
+        @input_validator = InputValidator.new
+    end
+
+    def welcome_player
         puts Prompt::WELCOME
-        CommandLineIn.new.get_input
         puts Prompt::MAKE_TOKEN_SELECTION
-        selected_token = CommandLineIn.new.get_input
-        puts "Thanks for selecting #{selected_token}. Let's start the game!" 
-        puts BoardPresenter.new.display_board
+        @@selected_token = CommandLineIn.new.get_input.upcase
+        if input_validator.valid_token?(@@selected_token)
+            puts "Thanks for selecting #{@@selected_token}. Let's start the game!"
+        else
+            welcome_player
+        end
+    end
+
+    def play
+        welcome_player
+        puts board_presenter.display_board(board.moves)
+        select_coordinate
+        take_turn(@@selected_token, @@selected_move)
+        puts board_presenter.display_board(board.moves)
+    end
+
+    def take_turn(token, position)
+        if input_validator.valid_coordinate?(position) && input_validator.valid_move?(board.moves, position)
```

Comment:
Both of these conditionals are really just trying to check the same thing. Whether or not the move made is valid. This can be summed up in one method instead of two.

Reply:
I'm checking for two things: the user is supplying an integer, the user is supplying an integer that has yet to be supplied during another turn.